test/Catch.hs

```diff
@@ -11,26 +11,24 @@ import Test.Tasty.Hedgehog
 
 gen
   :: forall e m a sig
-  .  (Has (Catch e) sig m, Arg e, Show a, Show e, Vary e)
+  .  (Has (Catch e) sig m, Arg e, Show e, Vary e)
   => Gen e
-  -> (forall a . Show a => Gen a -> Gen (With (m a)))
+  -> (forall a . Gen a -> Gen (m a))
   -> Gen a
-  -> Gen (With (m a))
-gen _ m a = do
-  h <- fn @e (m a)
-  subterm (m a) $ \ m -> liftWith2 "catchError" catchError m (fmap getWith <$> showingFn h)
+  -> Gen (m a)
+gen _ m a = label "catchError" catchError <*> m a <*> fn @e (m a)
 
 
 test
   :: forall e m a b sig
   .  (Has (Error e) sig m, Arg e, Eq a, Eq e, Show a, Show e, Vary e)
   => Gen e
-  -> (forall a . Show a => Gen a -> Gen (With (m a)))
+  -> (forall a . Gen a -> Gen (m a))
   -> Gen a
   -> Gen b
   -> (forall a . m a -> PureC (Either e a))
   -> [TestTree]
 test e m a _ runCatch =
   [ testProperty "catchError intercepts throwError" . forall (e :. fn (m a) :. Nil) $
-    \ e (FnWith h) -> runCatch (throwError e `catchError` h) === runCatch (h e)
+    \ e h -> runCatch (throwError e `catchError` h) === runCatch (h e)
```

Comment:
`fn` is now specialized to our new definition of `Gen`, and produces a (showable) function directly, rather than requiring any sort of unwrapping.